benchmarks/python/test_rope.py

```diff
@@ -176,3 +62,80 @@ def test_rope_benchmark(
 
     if not disable_benchmarking:
         run_benchmark(benchmark, fd.execute, inputs)
+
+
+@pytest.mark.parametrize(
+    "rope_variation",
+    [
+        "llama_2_7b_hf_rope",
+        "llama_3_8B_rope",
+        "hf_qwen2_rope",
+        "hf_phi3_rope",
+        "hf_mistral_nemo_rope",
+    ],
+)
+@pytest.mark.parametrize(
+    "executor", ["eager", "torchcompile", "thunder", "thunder-torchcompile"]
+)
+def test_rope_variations_fwd_benchmark(
+    benchmark,
+    rope_variation: str,
+    executor: str,
+):
+    kwargs = {}
+    if executor == "thunder":
+        kwargs["nv_enable_matmul"] = True
+    elif executor == "torchcompile":
+        clear_dynamo_cache()
+
+    model, inputs, _, _ = rope_setup[rope_variation]()
+
+    def fwd_call(inp):
+        return model(*inp)
+
+    # Compile the fwd fn for torchcompile
+    benchmark_fn = with_executor(executor, fwd_call, **kwargs)
+    run_benchmark(benchmark, benchmark_fn, inputs())
+
+
+@pytest.mark.parametrize(
+    "rope_variation",
+    [
+        "llama_2_7b_hf_rope",
+        "llama_3_8B_rope",
+        "hf_qwen2_rope",
+        "hf_phi3_rope",
+        "hf_mistral_nemo_rope",
+    ],
+)
+@pytest.mark.parametrize(
+    "executor", ["eager", "torchcompile", "thunder", "thunder-torchcompile"]
+)
+def test_rope_variations_bwd_benchmark(
+    benchmark,
+    rope_variation: str,
+    executor: str,
+):
+    kwargs = {}
+    if executor == "thunder":
+        kwargs["nv_enable_matmul"] = True
+    elif executor == "torchcompile":
+        clear_dynamo_cache()
+
+    model, fwd_inputs, grad, iobytes = rope_setup[rope_variation]()
+
+    def fwd_call(inp):
+        return model(*inp)
+
+    # execute the compiled fwd fn
+    fwd_fn = with_executor(executor, fwd_call, **kwargs)
+    outputs = fwd_fn(fwd_inputs())
+
+    # accumulate all output, so we can feed a single grad and use the unary bwd function
+    output = outputs[0]
+    for i in range(1, len(outputs)):
+        output += outputs[i]
+
+    # NOTE: the iobytes is computed based on how thunder autograd worked. So this is just
```

Comment:
I agree with your previous point about the IObytes computation potentially being different between the executors. I discussed this with @kevinstephano but off-hand I am not sure of a more robust way. I will open an issue to look into it. For this PR, the nvfuser-definition based computation LGTM since that is consistent with the other benchmarks.